src/Features/CSharp/Portable/ConvertLinq/ConvertForEachToLinqQuery/ToCountConverter.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.ConvertLinq.ConvertForEachToLinqQuery;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertLinq.ConvertForEachToLinqQuery
+{
+    /// <summary>
+    /// Provides a conversion to query.Count().
+    /// </summary>
+    internal sealed class ToCountConverter : AbstractToMethodConverter
+    {
+        public ToCountConverter(
+            ForEachInfo<ForEachStatementSyntax, StatementSyntax> forEachInfo,
+            ExpressionSyntax selectExpression,
+            ExpressionSyntax modifyingExpression,
+            SyntaxTrivia[] trivia)
+            : base(forEachInfo, selectExpression, modifyingExpression, trivia)
+        {
+        }
+
+        protected override string MethodName => nameof(Enumerable.Count);
+
+        // Checks that the expression is "0".
+        protected override bool CanReplaceInitialization(
+            ExpressionSyntax expression, 
+            CancellationToken cancellationToken)
+            => expression is LiteralExpressionSyntax literalExpression && literalExpression.Token.ValueText == "0";
```

Comment:
>expression is LiteralExpressionSyntax literalExpression && literalExpression.Token.ValueText == "0" [](start = 15, length = 99)

Why not [SemanticModel.GetConstantValue](http://source.roslyn.io/#Microsoft.CodeAnalysis/Compilation/SemanticModel.cs,731)?

Reply:
Using GetConstantValue for the case seems to be more complex both for execution and for readability. Not?

---
In reply to: [196974808](https://github.com/dotnet/roslyn/pull/27130#discussion_r196974808) [](ancestors = 196974808)